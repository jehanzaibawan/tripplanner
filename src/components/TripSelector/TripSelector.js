import React, { useState, useEffect } from "react";
import SearchableSelectBox from "../SearchableSelectBox/SearchableSelectBox";
import PropTypes from "prop-types";

const TripSelector = props => {
  const defaultValue = { value: "", label: "Select or type a keyword" };

  const [trip, setTrip] = useState({
    from: defaultValue,
    to: defaultValue
  });

  useEffect(() => {
    props.updateTripDirections(trip.from.label, trip.to.label);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [trip]);

  const selectBoxFromHandleChange = selectedItem => {
    setTrip({ ...trip, from: selectedItem });
  };

  const selectBoxToHandleChange = selectedItem => {
    setTrip({ ...trip, to: selectedItem });
  };

  const formatTripOptions = options => {
    if (Array.isArray(options) && options.length > 0)
      return options.map(option => ({
        value: option.toLowerCase(),
        label: option
      }));
    else return [];
  };

  return (
    <React.Fragment>
      <SearchableSelectBox
        id="select-box-from"
        label="From"
        options={formatTripOptions(props.options.departures)}
        value={trip.from}
        handleChange={selectBoxFromHandleChange}
        isOptionDisabled={option => option.value === trip.to.value}
      />
      <SearchableSelectBox
        id="select-box-to"
        label="To"
        options={formatTripOptions(props.options.arrivals)}
        value={trip.to}
        handleChange={selectBoxToHandleChange}
        isOptionDisabled={option => option.value === trip.from.value}
      />
    </React.Fragment>
  );
};

TripSelector.propTypes = {
  options: PropTypes.shape({}).isRequired,
  updateTripDirections: PropTypes.func.isRequired
};

export default TripSelector;
