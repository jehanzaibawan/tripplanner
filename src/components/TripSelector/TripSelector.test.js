import React from "react";
import renderer from "react-test-renderer";
import TripSelector from "./TripSelector";

it("renders a snapshot", () => {
  const props = {
    options: { departures: [], arrivals: [] },
    updateTripDirections: jest.fn()
  };
  const tree = renderer.create(<TripSelector {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
