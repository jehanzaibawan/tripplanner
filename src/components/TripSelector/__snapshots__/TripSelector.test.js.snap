// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
Array [
  <div
    className="makeStyles-root-1"
  />,
  <div
    className="makeStyles-root-1"
  />,
]
`;
